modules/videosipgw/JitsiVideoSIPGWSession.js

```diff
@@ -0,0 +1,152 @@
+/* global $, $iq */
+import { getLogger } from 'jitsi-meet-logger';
+const logger = getLogger(__filename);
+
+import EventEmitter from 'events';
+
+import * as Constants from './VideoSIPGWConstants';
+
+/**
+ * The event name for current sip video session state changed.
+ * @type {string} event name for sip video session state changed.
+ */
+const STATE_CHANGED = 'STATE_CHANGED';
+
+/**
+ * Jitsi video SIP GW session. Holding its state and able to start/stop it.
+ * When session is in OFF or FAILED stated it cannot be used anymore.
+ */
+export default class JitsiVideoSIPGWSession {
+
+    /**
+     * Creates new session with the desired sip address and display name.
+     *
+     * @param {string} sipAddress - The sip address to use when
+     * starting the session.
+     * @param {string} displayName - The display name to use for
+     * that participant.
+     * @param {ChatRoom} chatRoom - The chat room this session is bound to.
+     */
+    constructor(sipAddress, displayName, chatRoom) {
+        this.sipAddress = sipAddress;
+        this.displayName = displayName;
+        this.chatRoom = chatRoom;
+
+        this.eventEmitter = new EventEmitter();
+    }
+
+    /**
+     * Stops the current session.
+     */
+    stop() {
+        if (this.state === Constants.STATE_OFF
+            || this.state === Constants.STATE_FAILED) {
+            logger.warn('Video SIP GW session already stopped or failed!');
+
+            return;
+        }
+
+        this._sendJibriIQ('stop');
+    }
+
+    /**
+     * Starts a new session. Sends an iq to the focus.
+     */
+    start() {
+        // if state is off, this session was active for some reason
+        // and we should create new one, rather than reusing it
+        if (this.state === Constants.STATE_ON
```

Comment:
this check feels weird... so the initial value is "undefined" ? Can we initialise 'state' in the constructor and document it's default value ?